yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -353,7 +353,7 @@ private[yarn] class YarnAllocator(
 
     } else if (missing < 0) {
       val numToCancel = math.min(numPendingAllocate, -missing)
-      logInfo(s"Canceling requests for $numToCancel executor containers")
+      logInfo(s"Canceled requests for $numToCancel executor container(s)")
```

Comment:
The first message is the Spark Driver requesting the spark backend a total number of executors based on dynamic allocation.  The cancel message is coming in later because that is when the backend allocator actually figures out what it needs to ask YARN for.  They are different components. 

In this case we already have the requested number running or pending to be allocated (from YARN) so we actually have to cancel some of those reqeusts.  We haven't canceled them yet, we are about to, so it shouldn't be changed to canceled.

If there is a way to update the log messages to be more clear I"m ok with that, otherwise this should be closed.